angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
I agree with respect to PR motivation. I think the type of this function doesn't need this level of specificity.

Thank you for explaining the `length` behavior.

That said, even when used from html, treating strings like arrays because they have this property is a bit odd. I've used `ng-repeat` thousands of times and never intentionally passed a string.